app/cdap/components/AddTransformation/SelectMultipleColumnList/index.tsx

```diff
@@ -0,0 +1,273 @@
+import {
+  Box,
+  Checkbox,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  FormControlLabel,
+  Typography,
+} from '@material-ui/core';
+import React, { useEffect, useRef, useState } from 'react';
+import { useStyles } from '../styles';
+import { SearchIcon } from '../iconStore';
+import { prepareDataQualtiy } from '../CircularProgressBar/utils';
+import DataQualityProgress from '../CircularProgressBar';
+import { NoDataSVG } from 'components/GridTable/iconStore';
+import { multipleColumnSelected } from '../constants';
+import T from 'i18n-react';
+
+const SelectColumnsList = (props) => {
+  const {
+    directiveFunctionSupportedDataType,
+    selectedColumnsCount,
+    columnData,
+    setSelectedColumns,
+    dataQuality,
+    functionName,
+  } = props;
+  const [columns, setColumns] = useState(columnData);
+  const [dataQualityValue, setDataQualityValue] = useState(dataQuality);
+  const [selectedColumns, setSelectedColumn] = useState([]);
+  const [focused, setFocused] = useState(false);
+  const classes = useStyles();
+  const ref = useRef(null);
+  const no_match =
+    directiveFunctionSupportedDataType.length > 0
+      ? !directiveFunctionSupportedDataType.includes('all')
+      : false
+      ? columns.filter((object1) => {
+          return directiveFunctionSupportedDataType.some((object2) => {
+            return object2.includes(object1.type[0].toLowerCase());
+          });
+        })
+      : [];
+  useEffect(() => {
+    const getPreparedDataQuality = prepareDataQualtiy(dataQuality, columnData);
```

Comment:
i think the code in this component is repeating
app/cdap/components/AddTransformation/SelectColumnsList/index.tsx

can we make this as a reusable component and reuse based on conditions rather than writing multiple components for same functionality or usage

Reply:
Done